### DISCUSSION TITLE
QZ speed sensor data over ANT+

### DISCUSSION THREAD

**Discussion by oel59** [cagnulein/qdomyos-zwift#2335] [General]
Hi,

I enabled the ANT+ Cadence option in the QZ settings.
Thanks to this feature I can now connect QZ to my Garmin watch as PWR and Spd/Cad sensors.
 My Domyos bike can connect directly to Kinomap without need for QZ, but in a new set-up I connected the Domyos bike to QZ (As a Wahoo Trainer). The problem with Kinomap is, that it does not upload its data to Garmin Connect, and even if you manage to upload a TCX file manually, Garmin does not take the data into account for the Performance Data.
With the new set up, I start an Indoor Bike activity on the watch, in parallel with a Kinomap session. My watch will now receive power and cadence data from QZ and together with the Heart rate data already in the watch, it producces good data for the Performance statistics in Garmin Connect.

The only data missing now in Garmin Connect is the speed data, and I have to manually enter the distance in the activity to get average speed.

Now my question: Would it be possible for QZ to send the speed data over ANT+ in addition to the Power and Cadence data that is already sent? I assume that the Spd/Cad sensor on the watch will pick up that data?

Thanks for letting me know your thoughts on this.

**Comment by cagnulein** [comment]
hi @oel59 i'm already sending speed with cadence and power. did you set the right diamater millimiter on the watch?

**Reply by oel59** [reply]
No, I did not do that. To be honest I did not understand that fully. 
What value should I put there?

**Reply by cagnulein** [reply]
100mm for the wheel diameter

**Reply by oel59** [reply]
Thank you very much Roberto.

Have a nice day,

Ciao,

Leo.

**Reply by oel59** [reply]
Sorry Roberto, 

10 cm for a wheel diameter? Should it not be 1000mm?
Or maybe I misunderstand the concept here?

Thanks.

**Reply by cagnulein** [reply]
nope set 100mm (10cm), it's the default of QZ

**Comment by oel59** [comment]
Ok

**Comment by oel59** [comment]
Hi Roberto,

I have set the wheel size in the garmin watch for the cad/Spd sensor, but
still don't get the speed in the watch.

Maybe it is a problem on the watch side?

Kind regards,

Leo.

**Comment by cagnulein** [comment]
But do you see the cadence correctly? The watch should just multiply it for
the diameter.
Maybe can you make me a video?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mar 14 mag 2024 alle 18:52 oel59 ***@***.***> ha
scritto:

> Hi Roberto,
>
> I have set the wheel size in the garmin watch for the cad/Spd sensor, but
> still don't get the speed in the watch.
>
> Maybe it is a problem on the watch side?
>
> Kind regards,
>
> Leo.
>
>
>

**Reply by oel59** [reply]
Hi,

Yes, I see the cadence correctly. I had attached a picture to my previous reply. Will attach it again, but I can try to send more tomorrow.

Thanks,

Leo.
![20240514_184555](https://github.com/cagnulein/qdomyos-zwift/assets/127523932/74ea50a9-0046-411f-9e41-3e5f2ee0c7b8)

**Reply by cagnulein** [reply]
Yeah I would like to see the sensor settings

**Reply by oel59** [reply]
![20240514_211030](https://github.com/cagnulein/qdomyos-zwift/assets/127523932/88faef01-53cd-4b5d-9d05-f86b5494824b)

**Reply by cagnulein** [reply]
Is it under the spd cadence sensor? I will also test this on my forerunner

**Comment by oel59** [comment]
Hi,

Yes it is.

**Comment by oel59** [comment]
Ciao Roberto,

I saw on a Garmin forum that some Garmin devices don't take the speed from
a sensor if GPS is deactivated, which would be the case for an indoor bike
activity.
Have you been able to see what happens with your Forerunner?

Thanks,

Kind regards,

Leo.

**Comment by cagnulein** [comment]
Not yet, anyway it's working for a Fenix 6 here for example #2285 so I
don't think it's the case.
I will update you when I will test it


Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 15 mag 2024 alle 23:30 oel59 ***@***.***> ha
scritto:

> Ciao Roberto,
>
> I saw on a Garmin forum that some Garmin devices don't take the speed from
> a sensor if GPS is deactivated, which would be the case for an indoor bike
> activity.
> Have you been able to see what happens with your Forerunner?
>
> Thanks,
>
> Kind regards,
>
> Leo.
>

**Comment by cagnulein** [comment]
Just checked and it works on the forerunner 255 too

![image](https://github.com/cagnulein/qdomyos-zwift/assets/48216/5da55c05-96d8-426d-9e6a-4debc95ca133)
![image](https://github.com/cagnulein/qdomyos-zwift/assets/48216/eb2e8729-e3fa-4cd1-93de-4b4b6ea573e0)
![image](https://github.com/cagnulein/qdomyos-zwift/assets/48216/25138e5f-97ba-463d-91c8-304a8a42d842)

**Comment by oel59** [comment]
Thanks Roberto,

I have a Marq Golfer, but this watch has the Fenix6 software on it.
I really don't understand why I don't see the speed in my watch.
Anyway, it is not that important information, I am happy with the power and
cadence data.

Have a great day,

Leo.

**Comment by cagnulein** [comment]
There is for sure an explanation. Maybe a factory reset could fix this

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 16 mag 2024 alle 07:36 oel59 ***@***.***> ha
scritto:

> Thanks Roberto,
>
> I have a Marq Golfer, but this watch has the Fenix6 software on it.
> I really don't understand why I don't see the speed in my watch.
> Anyway, it is not that important information, I am happy with the power
> and
> cadence data.
>
> Have a great day,
>
> Leo.
>

**Reply by oel59** [reply]
I noticed in your pictures that you set 1000mm and not 100mm as you suggested earlier. I will try that.

**Comment by cagnulein** [comment]
My bad, I just woke up and I put it wrong, anyway it should work in both
case, you will see a different speed value of course

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 16 mag 2024 alle 07:40 oel59 ***@***.***> ha
scritto:

**Comment by oel59** [comment]
Hi Roberto,

Just to give you an update:

Yesterday I received a software update for my MARQ Garmin watch.

This morning huge problems with ANT+
QZ did not receive the broadcasted heart rate and the watch could no longer
connect to the QZ sensors.

I removed the sensors from the watch, restarted the phone and the watch
then reconnected the sensors, and.......everything worked.

Amazing.

**Reply by cagnulein** [reply]
I got a Forerunner for my wife 2 weeks ago and I saw a lot of connectivy bugs on the garmin software so your story doesn't surprise me at all unfortunately.
Anyway glad that it's working now!

**Reply by oel59** [reply]
And thanks for your help as always
Have a nice day

Leo
